platform/common/services/sig/service.go

```diff
@@ -60,36 +61,46 @@ func (o *Service) RegisterSigner(identity view.Identity, signer driver.Signer, v
 		return errors.New("invalid signer, expected a valid instance")
 	}
 
+	// check existence with a read lock
 	idHash := identity.UniqueID()
-	o.viewsSync.Lock()
+	o.mutex.RLock()
 	s, ok := o.signers[idHash]
-	o.viewsSync.Unlock()
+	o.mutex.RUnlock()
+	if ok {
+		logger.Infof("another signer bound to [%s]:[%s][%s] from [%s]", identity, GetIdentifier(s), GetIdentifier(signer), string(s.DebugStack))
```

Comment:
when does this happen? I saw a lot of those in the logs (used to be warn even) and always wondered if it was part of the normal flow.

Reply:
I did some exploration in past, I can check again

Reply:
we can define put it as debug though